### ISSUE TITLE
Improve UX re: NaNs and hyper-parameters

### ISSUE THREAD

**Issue by bm371613** [mim-solutions/embedding-sum#4]
The model can produce NaNs if the hyper parameters (like learning rate) is too high (potentially also if the input data contains NaNs?)

It would be nice it this was detected and a helpful message would suggest a solution.
